src/ibex_bluesky_core/fitting.py

```diff
@@ -687,3 +688,58 @@ def guess(
             return init_guess
 
         return guess
+
+
+class MuonMomentum(Fit):
+    """Muon momentum fitting."""
+
+    equation = """
+        y=(erfc((x-x0/w))*(r/2)+b)*((x/x0)**p)"""
+
+    @classmethod
+    def model(cls, *args: int) -> lmfit.Model:
+        """Momentum scan model."""
+
+        def model(
+            x: npt.NDArray[np.float64], x0: float, r: float, w: float, p: float, b: float
+        ) -> npt.NDArray[np.float64]:
+            return (scipy.special.erfc((x - x0) / w) * (r / 2) + b) * ((x / x0) ** p)
+
+        return lmfit.Model(model, name=f"{cls.__name__}  [{cls.equation}]")
+
+    @classmethod
+    def guess(
+        cls, *args: int
+    ) -> Callable[[npt.NDArray[np.float64], npt.NDArray[np.float64]], dict[str, lmfit.Parameter]]:
+        """Momentum Scan Fit Guessing."""
+
+        def guess(
+            x: npt.NDArray[np.float64], y: npt.NDArray[np.float64]
+        ) -> dict[str, lmfit.Parameter]:
+            min_y = np.argmin(y)
+            max_y = np.argmax(y)
+
+            b = y[min_y]
+            r = y[max_y] - b
+
+            x_slope = x[max_y:min_y]  # Gets all x values between the maximum and minimum y
+
+            if len(x_slope) != 0:
+                x0 = np.mean(x_slope)
+            else:
+                x0 = x[-1]  # Picked as it can't be 0
+
+            p = 1  # Expected value, not likely to change
+
+            w = (x[-1] - x[0]) / 9  # 9 is a rough estimate base on testing
```

Comment:
I think we should pull out magic numbers like this `9` here as constants - even if just as a variable above this statement.